salt/cloud/clouds/vmware.py

```diff
@@ -1448,7 +1524,7 @@ def _get_hba_type(hba_type):
     raise ValueError('Unknown Host Bus Adapter Type')
 
 
-def test_vcenter_connection(kwargs=None, call=None):
```

Comment:
Why are we removing kwargs?

These are required for calling stuff from the commandline using `salt-cloud -f` or `salt-cloud -a`

Reply:
Because kwargs aren't used in the function itself. I oscillated between putting pylint hints in place and removing kwargs when they're not used *in* a function, eventually settling with the latter for clarity.

> the code already meets our pylint style guidelines

Would you care to explain why running pylint with the `.pylintrc` (https://github.com/saltstack/salt/blob/develop/.pylintrc) in accompanying the SaltStack code spits out warnings and errors for things corrected? If the SaltStack Coding Style is not completely erroneous, I believe the pylint fixes introduced are in line with the coding standards and the code didn't meet the pylint style guidelines.

Reply:
We use .testing.pylintrc for our pylint tests, and it is currently passing on this branch.

https://jenkinsci.saltstack.com/job/develop/view/Lint/job/salt-develop-lint/

Reply:
In that case I suggest the relevant documentation would be amended to reflect that fact. Currently the documentation claims: "SaltStack recommends running the .pylintrc file on any files you are changing with your code contribution before submitting a pull request to Salt's repository."

Mentioning that against the branch "development" a separate pylintrc is being used would help to avoid misunderstandings in the future.

... that said, I still stand by my fabulous argument about not putting unused arguments in the function fingerprints (obviously I did not remove kwargs or any other parameters from functions which actually use it).

Reply:
@samilaine is right on here about the error on our docs. That's been there for years and I keep forgetting that it needs to be cleaned up.

Reply:
Yeah, that needs to be cleaned up, over the last couple weeks, we have moved pylint to use `tox` so to use the same lint command that we are using, you can run `tox -e pylint-salt,pylint-tests`  You can also pass a noops operator, and pass what files you want to lint afterwards, and it should only test those files.

```
tox -e pylint-salt -- salt/cloud/clouds/vmware.py
```

As for removing kwargs, that still needs to be in there because of the way that we call those functions when we call them inside of the CloudClient (if someone passes in kwargs) when calling functions and actions.

https://github.com/saltstack/salt/blob/develop/salt/cloud/__init__.py#L1610
https://github.com/saltstack/salt/blob/develop/salt/cloud/__init__.py#L1534